### ISSUE TITLE
opm: index catalog size doubles when adding bundles from index

### ISSUE THREAD

**Issue by jdockter** [operator-framework/operator-registry#259]
Related to issue #218
When running the following commands it looks like the size of the index catalog image is doubling:

```
opm index add --bundles quay.io/jondockter/memcached-operator-bundle:0.0.1 --tag quay.io/jondockter/test-catalog-index:1.0.0
```

```
opm index add --bundles quay.io/jondockter/memcached-operator-bundle:0.0.2 --from-index quay.io/jondockter/test-catalog-index:1.0.0 --tag quay.io/jondockter/test-catalog-index:1.0.1
```

podman images
```
REPOSITORY                                             TAG      IMAGE ID       CREATED         SIZE
<none>                                                 <none>   4ffe9f324d60   4 minutes ago   2.52 GB
quay.io/jondockter/test-catalog-index                  1.0.1    228e1b258e23   4 minutes ago   2.52 GB
<none>                                                 <none>   427d23d38fe5   6 minutes ago   1.26 GB
quay.io/jondockter/test-catalog-index                  1.0.0    345aec45d18f   6 minutes ago   1.26 GB
quay.io/jondockter/memcached-operator-bundle           0.0.3    46f7586bbaaa   6 minutes ago   17.5 kB
quay.io/jondockter/memcached-operator-bundle           0.0.2    799af2276c92   6 minutes ago   17.5 kB
quay.io/jondockter/memcached-operator-bundle           0.0.1    76f66e95a108   7 minutes ago   17.5 kB
quay.io/operator-framework/upstream-registry-builder   latest   53136ea017f3   23 hours ago    1.26 GB
```

fyi... @lcarva

**Comment by nathanbrophy** [comment]
I ran a `docker save` on the images being made from the opm commands and in the layer that contains the database for the first image made (not made from existing catalog) I see the following:

```
image/
  db_layer/
    database/
      index.db
```
This is what I would expect to see.  If I create an image using a previous catalog image (adding a bundle to an existing catalog) I see the following:

```
mage/
  db_layer/
    database/
      database/
      src/
      etc/
      .... # There is an entire copy of previous catalog image in the database folder
```

I can produce this with the following commands using release 1.7 (latest release as of today) of opm

```
$ operator-sdk bundle create "quay.io/<namespace>/operator-name:1.0.0" --directory "$OLMCATALOGPATH/$PACKAGENAME/$VERSION" --package "operatorbundle" --channels "beta" --default-channel "beta"

# Make bundle 1

$ operator-sdk bundle create "quay.io/<namespace>/operator-name:1.0.1" --directory "$OLMCATALOGPATH/$PACKAGENAME/$VERSION" --package "operatorbundle" --channels "beta" --default-channel "beta"

# Make bundle 2

$ opm index add -b "$quay.io/<namespace>/operator-name:1.0.0" -c "docker|podman" --tag "quay.io/<namespace>/testcatalog:latest" --debug

# Initialize the catalog source

$ opm index add -b "$quay.io/<namespace>/operator-name:1.0.1" -c "docker|podman" --tag "quay.io/<namespace>/testcatalog:latest" --debug

# Add another bundle to the catalog source
```

**Comment by lcarva** [comment]
I can't run the example provided in the description with opm 1.17, due to:
```
"error loading bundle from image: Error adding package error loading bundle into db: table api_provider has no column named channel_entry_id"
```

Full logs:
```
🐚  opm-1.17 index add --bundles quay.io/jondockter/memcached-operator-bundle:0.0.1 --tag quay.io/jondockter/test-catalog-index:1.0.0
INFO[0000] building the index                            bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.1]"
INFO[0000] running podman pull                           img="quay.io/jondockter/memcached-operator-bundle:0.0.1"
INFO[0003] running podman save                           img="quay.io/jondockter/memcached-operator-bundle:0.0.1"
INFO[0003] loading Bundle quay.io/jondockter/memcached-operator-bundle:0.0.1  img="quay.io/jondockter/memcached-operator-bundle:0.0.1"
INFO[0003] found annotations file searching for csv      dir=bundle_tmp090825149 file=bundle_tmp090825149/metadata load=annotations
INFO[0003] found csv, loading bundle                     dir=bundle_tmp090825149 file=bundle_tmp090825149/manifests load=bundle
INFO[0003] loading bundle file                           dir=bundle_tmp090825149/manifests file=cache.example.com_memcacheds_crd.yaml load=bundle
INFO[0003] loading bundle file                           dir=bundle_tmp090825149/manifests file=memcached-operator.v0.0.1.clusterserviceversion.yaml load=bundle
INFO[0004] Generating dockerfile                         bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.1]"
INFO[0004] writing dockerfile: index.Dockerfile135483639  bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.1]"
INFO[0004] running podman build                          bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.1]"
INFO[0004] [podman build -f index.Dockerfile135483639 -t quay.io/jondockter/test-catalog-index:1.0.0 .]  bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.1]"
lucarval@localhost ~ 
(0) 🐚  opm-1.17 index add --bundles quay.io/jondockter/memcached-operator-bundle:0.0.2 --from-index quay.io/jondockter/test-catalog-index:1.0.0 --tag quay.io/jondockter/test-catalog-index:1.0.1
INFO[0000] building the index                            bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0000] Pulling previous image quay.io/jondockter/test-catalog-index:1.0.0 to get metadata  bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0000] running podman pull                           bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0001] Getting label data from previous image        bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0001] running podman inspect                        bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0001] running podman pull                           bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0002] running podman save                           bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]"
INFO[0013] running podman pull                           img="quay.io/jondockter/memcached-operator-bundle:0.0.2"
INFO[0015] running podman save                           img="quay.io/jondockter/memcached-operator-bundle:0.0.2"
INFO[0015] loading Bundle quay.io/jondockter/memcached-operator-bundle:0.0.2  img="quay.io/jondockter/memcached-operator-bundle:0.0.2"
INFO[0015] found annotations file searching for csv      dir=bundle_tmp769265966 file=bundle_tmp769265966/metadata load=annotations
INFO[0015] found csv, loading bundle                     dir=bundle_tmp769265966 file=bundle_tmp769265966/manifests load=bundle
INFO[0015] loading bundle file                           dir=bundle_tmp769265966/manifests file=cache.example.com_memcacheds_crd.yaml load=bundle
INFO[0015] loading bundle file                           dir=bundle_tmp769265966/manifests file=memcached-operator.v0.0.2.clusterserviceversion.yaml load=bundle
FATA[0015] permissive mode disabled                      bundles="[quay.io/jondockter/memcached-operator-bundle:0.0.2]" error="error loading bundle from image: Error adding package error loading bundle into db: table api_provider has no column named channel_entry_id"
```

**Comment by jdockter** [comment]
Looks like this was fixed in [1.12.1](https://github.com/operator-framework/operator-registry/releases/tag/v1.12.1)
